programs/marginfi/src/state/price.rs

```diff
@@ -111,10 +111,13 @@ impl OraclePriceFeedAdapter {
         match bank_config.oracle_setup {
             OracleSetup::None => Err(MarginfiError::OracleNotSetup.into()),
             OracleSetup::PythLegacy => {
-                check!(ais.len() == 1, MarginfiError::InvalidOracleAccount);
+                check!(
+                    ais.len() == 1,
+                    MarginfiError::PythLegacyWrongNumberOfAccounts
```

Comment:
I think it would be enough to have one "Wrong number of Accounts" errors for all the oracle types, one per type is probably overkill since you can easily read which type you have from the bank.